common/src/main/java/com/wynntils/features/inventory/HightlightDuplicateCosmeticsFeature.java

```diff
@@ -0,0 +1,151 @@
+
+    private final List<Component> selectedCosmetics = new ArrayList<>();
+
+    private boolean onScrapMenu = false;
+    private Component hoveredCosmetic;
+
+    @SubscribeEvent
+    public void onScreenInit(ScreenInitEvent event) {
+        if (!(event.getScreen() instanceof AbstractContainerScreen<?> screen)) return;
+        if (!(screen.getMenu() instanceof ChestMenu)) return;
+
+        StyledText title = StyledText.fromComponent(screen.getTitle());
+
+        SearchableContainerType searchableContainerType = SearchableContainerType.getContainerType(title);
+        if (searchableContainerType != SearchableContainerType.SCRAP_MENU) return;
+
+        onScrapMenu = true;
+    }
+
+    @SubscribeEvent
+    public void onScreenClose(ScreenClosedEvent e) {
+        selectedCosmetics.clear();
+        onScrapMenu = false;
+        hoveredCosmetic = null;
+    }
+
+    @SubscribeEvent
+    public void onContainerRender(ContainerRenderEvent event) {
+        if (!onScrapMenu) return;
+        // No need to get hovered cosmetic if only displaying selected cosmetics
+        if (highlightCondition.get() == HighlightCondition.SELECTED) return;
+
+        Slot hoveredSlot = event.getHoveredSlot();
+
+        // If no item or on new reward screen don't get a hovered cosmetic, hide §f too as the glass panes on new
+        // reward screen are named that and the return text item isn't added immediately
+        if (hoveredSlot == null
+                || hoveredSlot.getItem().getItem() == Items.AIR
+                || event.getScreen()
+                        .getMenu()
+                        .slots
+                        .get(RETURN_SLOT)
+                        .getItem()
+                        .getHoverName()
+                        .equals(RETURN_TEXT)
+                || hoveredSlot.getItem().getHoverName().equals(Component.literal("§f"))) {
+            hoveredCosmetic = null;
+            return;
+        }
+
+        hoveredCosmetic = hoveredSlot.getItem().getHoverName();
+    }
+
+    @SubscribeEvent
+    public void onRenderSlot(SlotRenderEvent.Pre e) {
+        if (!onScrapMenu) return;
+
+        // Don't highlight the cosmetics in the selected slots
+        if (SearchableContainerType.SCRAP_MENU.getBounds().getSlots().contains(e.getSlot().index)) {
+            if (selectedCosmetics.contains(e.getSlot().getItem().getHoverName())
+                    && highlightCondition.get() != HighlightCondition.HOVER) {
+                RenderUtils.drawArc(
+                        e.getPoseStack(), selectedHighlightColor.get(), e.getSlot().x, e.getSlot().y, 200, 1f, 6, 8);
+            } else if (e.getSlot().getItem().getHoverName().equals(hoveredCosmetic)
+                    && highlightCondition.get() != HighlightCondition.SELECTED) {
+                RenderUtils.drawArc(
+                        e.getPoseStack(), hoveredHighlightColor.get(), e.getSlot().x, e.getSlot().y, 200, 1f, 6, 8);
+            }
```

Comment:
The render call is the same, only the color changes. So that makes it easy to clean up this whole if/else chain.